File: go-client-sdk/pkg/models/operations/getonecrmcontact.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmContactQueryParams struct {
	AccessToken string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool   `queryParam:"style=form,explode=true,name=allFields"`
	Email       *string `queryParam:"style=form,explode=true,name=email"`
	ID          *string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmContactResponseBody struct {
	Contact *shared.Contact `json:"contact,omitempty"`
}

type GetOneCrmContactRequest struct {
	QueryParams GetOneCrmContactQueryParams
}

type GetOneCrmContactResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmContactResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getonecrmcontact.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmContactQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=email" })
  email?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id?: string;
}


export class GetOneCrmContactResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=contact" })
  contact?: shared.Contact;
}


export class GetOneCrmContactRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmContactQueryParams;
}


export class GetOneCrmContactResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmContactResponseBody;
}
